scripts/dc2/create_sne_ic.py

```diff
@@ -0,0 +1,132 @@
+import os
+import pandas as pd
+import numpy as np
+import gzip
+import shutil
+import copy
+import json
+import argparse
+from sqlalchemy import create_engine
+from lsst.utils import getPackageDir
+from lsst.sims.photUtils import Bandpass
+from lsst.sims.catUtils.supernovae import SNObject
+from lsst.sims.catUtils.utils import ObservationMetaDataGenerator
+from desc.sims.GCRCatSimInterface import get_obs_md
+
+
+class lensedSneCat():
+
+    def __init__(self, truth_cat, sed_path, write_sn_sed=True):
+
+        self.truth_cat = truth_cat
+        self.sn_obj = SNObject(0., 0.)
+        self.imSimBand = Bandpass()
+        self.imSimBand.imsimBandpass()
+        self.sed_path = sed_path
+        self.write_sn_sed = write_sn_sed
+
+        if not os.path.exists(self.sed_path):
+            os.mkdir(self.sed_path)
+
+    def calc_sne_mags(self, obs_mjd, obs_filter):
+
+        wavelen_max = 1800.
+        wavelen_min = 30.
+        wavelen_step = 0.1
+
+        sn_magnorm_list = []
+        sn_sed_names = []
+        add_to_cat_list = []
+
+        for idx in range(len(self.truth_cat)):
+
+            sn_param_dict = copy.deepcopy(self.sn_obj.SNstate)
+            sn_param_dict['_ra'] = np.radians(self.truth_cat['ra'].iloc[idx])
+            sn_param_dict['_dec'] = np.radians(self.truth_cat['dec'].iloc[idx])
+            sn_param_dict['z'] = self.truth_cat['redshift'].iloc[idx]
+            sn_param_dict['c'] = self.truth_cat['c'].iloc[idx]
+            sn_param_dict['x0'] = self.truth_cat['x0'].iloc[idx]
+            sn_param_dict['x1'] = self.truth_cat['x1'].iloc[idx]
+            sn_param_dict['t0'] = self.truth_cat['t0'].iloc[idx]
+
+            current_sn_obj = self.sn_obj.fromSNState(sn_param_dict)
+            current_sn_obj.mwEBVfromMaps()
+            sed_mjd = obs_mjd - self.truth_cat['t_delay'].iloc[idx]
+
+            sn_sed_obj = current_sn_obj.SNObjectSED(time=sed_mjd,
+                                                    wavelen=np.arange(wavelen_min, wavelen_max,
+                                                                      wavelen_step))
+            flux_500 = sn_sed_obj.flambda[np.where(sn_sed_obj.wavelen >= 499.99)][0]
+
+            if flux_500 > 0.:
+                sn_magnorm = current_sn_obj.catsimBandMag(self.imSimBand, sed_mjd)
+                sn_name = None
+                if self.write_sn_sed:
+                    sn_name = 'specFileGLSN_%i_%i_%.4f.txt' % (self.truth_cat['dc2_sys_id'].iloc[idx],
```

Comment:
The code that parses the instance catalogs looks for SED files either in `$SIMS_SED_LIBRARY_DIR` or in the directory where the instance catalog file lives.  So I think this line should be
```
sn_name = 'Dynamic/specFileGLSN_%i_%i_%.4f.txt' % (self.truth_cat['dc2_sys_id'].iloc[idx],
```
(The folder can be something other than `Dynamic`, but that's what the not-strongly lensed SNe code uses.)
Similarly, line 75 below should be
```
sn_sed_names.append(sn_name + '.gz')
```
Note that the `.gz` also needs to be added.

With these changes an instance catalog entry would look like
```
object GLSN_2096_1 52.909090 -28.130903 24.737524 Dynamic/specFileGLSN_2096_1_60608.1487.txt.gz 0.374896 0 0 0 0 0 point none CCM 0.028191 3.100000
```
and this script would be called something like this:
```
visit_id=696292
visit_dir="$(printf "%08d" $visit_id)"
out_dir=Run3.0_instcats
python create_sne_ic.py \
    --obs_db /global/projecta/projectdirs/lsst/groups/SSim/DC2/minion_1016_desc_dithered_v4.db \
    --obs_id $visit_id --sne_truth_cat $sne_truth_cat --file_out $out_dir/$visit_dir/lensed_sne_$visit_id.txt \
    --sed_out $out_dir/$visit_dir
```

Reply:
Ok, updated the instance catalog files to get this in line with what's expected.

Ran it with:
`bash write_instance_catalogs.bash 230 Run3.0_instcats /global/cscratch1/sd/brycek/example_truth/agn_truth.db /global/cscratch1/sd/brycek/example_truth/sne_truth.db Dynamic`